Shared/Realm.Shared/RealmObject.cs

```diff
@@ -42,7 +42,7 @@ public class RealmObject : IReflectableType, INotifyPropertyChanged
         #region static
 
         #if __IOS__
-        [MonoPInvokeCallback(typeof(NativeCommon.NotifyRealmCallback))]
+        [MonoPInvokeCallback(typeof(NativeCommon.NotifyRealmObjectCallback))]
```

Comment:
Should we fix that in a separate PR to keep a cleaner history, or is it insignificant enough? @kristiandupont @fealebenpae